File: ui/src/types/category.ts
export interface Category {
  label: string;
}

export interface CategoryStoreState {
  loading: boolean;
  categories: Category[];
  default: string;
}

export const InitialCategoryStoreState: CategoryStoreState = {
  loading: false,
  categories: [],
  default: "Algorithms",
};

File: v2/_api/src/main/java/com/studyseeking/models/Category.java
package com.studyseeking.models;

import lombok.*;

import javax.persistence.*;

@Entity
@Table(name = "categories")
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Getter
@Setter
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "application_id", nullable = false)
    private Application application;

    @Column(name = "label", nullable = false)
    private String label;
}
